src/modules/welcome/WelcomePage.cpp

```diff
@@ -77,11 +77,25 @@ WelcomePage::WelcomePage( Config* conf, QWidget* parent )
     ui->aboutButton->setIcon( CalamaresUtils::defaultPixmap(
         CalamaresUtils::Information,
         CalamaresUtils::Original,
-        2 * QSize( CalamaresUtils::defaultFontHeight(), CalamaresUtils::defaultFontHeight() ) ) );
+        2 * QSize( defaultFontHeight, defaultFontHeight ) ) );
     connect( ui->aboutButton, &QPushButton::clicked, this, &WelcomePage::showAboutBox );
 
+    // insert system-check widget below welcome text
     int welcome_text_idx = ui->verticalLayout->indexOf( ui->mainText );
     ui->verticalLayout->insertWidget( welcome_text_idx + 1, m_checkingWidget );
+
+    // insert optional logo banner image above welcome text
+    Calamares::Branding::ImageEntry bannerImage = Calamares::Branding::ProductBanner;
+    QString bannerPath = Calamares::Branding::instance()->imagePath( bannerImage );
```

Comment:
I think you should add a `if (!bannerPath.isEmpty()) {}` around the next block, if only to avoid typing-to-load-a-QImage that is obviously bogus.

Reply:
the problem with checking the string, is that the image file may not be present, regardless of whether the branding key is defined or not - so if not checking for a null image, it would need instead to check for a nonexistent file, before instantiating an image

Reply:
That makes sense -- my instinct was "if it's not set, don't do anything" but of course if it's not set, then the image-loading bits won't trigger and your code already does the right thing there. So it's good.